From 630d3ae167a51d3ae5b78eda4aadd053e5b838e6 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 15:58:59 +0300
Subject: [PATCH] Revert "[SW-82716] gaudi3: add pcie device id api"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit 3b6ddd1dd518fc241482df16e68e0d0b21d4a91a.

Change-Id: Ib337203344f80f4f98460f64e254e795f729163c
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 31e7904..e85a85d 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -919,21 +919,6 @@
 	arcboot_rom_set_spi_tpm_cs_gpio(cs_en);
 }
 
-uint32_t arcboot_rom_get_device_id(void)
-{
-	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_SECURITY_ENABLE_MASK) {
-		if (arcboot_rom_global.sku)
-			return PCI_IDS_GAUDI3_SINGLE_DIE_SEC;
-		else
-			return PCI_IDS_GAUDI3_SEC;
-	} else {
-		if (arcboot_rom_global.sku)
-			return PCI_IDS_GAUDI3_SINGLE_DIE;
-		else
-			return PCI_IDS_GAUDI3;
-	}
-}
-
 void fuse_prog_range_config(void)
 {
 	regmap_writel(D0_PSOC_DFT_EFUSE_PROG_ADDRESS_LAST_M, mmD0_PSOC_EFUSE_PROG_ADDRESS);
diff --git a/include/arcboot.h b/include/arcboot.h
index 9fc9162..54ca303 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -289,7 +289,6 @@
 uint32_t arcboot_error_read_and_clr(uint32_t *major, uint32_t *minor);
 bool arcboot_error_inform(uint32_t major, uint32_t minor);
 bool arcboot_error_is_pending(void);
-uint32_t arcboot_rom_get_device_id(void);
 
 #define CLEAR_DMA_SECURITY_PROGRESS 0x3
 
